eng/pipelines/templates/stages/archetype-cpp-release.yml

```diff
@@ -86,53 +86,93 @@ stages:
                     steps:
                       - checkout: self
 
+                      - pwsh: |
+                          $branchName = "azure-sdk-for-cpp-${{ artifact.Name }}-$(Build.BuildId)"
+                          if ($env:VCPKGFORKBRANCHNAME) { 
+                            Write-Host "Using queue time branch name" 
+                            $branchName = $env:VCPKGFORKBRANCHNAME
+                          } 
+                          Write-Host "##vso[task.setvariable variable=PrBranchName]$branchName"
+                        displayName: Set fork branch name
+
                       - task: Powershell@2
                         inputs:
                           filePath: eng/scripts/Initialize-VcpkgRelease.ps1
                           arguments: >-
-                            -GitHubPat $(azuresdk-github-pat)
                             -GitHubRepo "$(Build.Repository.Name)"
                             -PackageSpecPath $(Pipeline.Workspace)/packages/${{artifact.Name}}/package-info.json
                             -SourceDirectory $(Pipeline.Workspace)/packages/${{artifact.Name}}/vcpkg
                             -Verbose
                           pwsh: true
                         displayName: Initialize vcpkg release
 
+                      # There are potential race conditions if this script runs
+                      # in parallel against the same branch name. Release only 
+                      # one package at a time.
                       - pwsh: |
+                          Write-Host "git clone https://github.com/microsoft/vcpkg $(Pipeline.Workspace)/vcpkg"
                           git clone https://github.com/microsoft/vcpkg $(Pipeline.Workspace)/vcpkg
                           if ($LASTEXITCODE -ne 0) {
                             Write-Error "Unable to check out vcpkg repo"
-                            exit $LASTEXITCODE
+                            exit $LASTEXITCODE`
                           }
+                          
+                          Write-Host "cd $(Pipeline.Workspace)/vcpkg"
                           cd $(Pipeline.Workspace)/vcpkg
-
-                          # There is no need to check out a branch, the
-                          # create-pull-request.yml template creates the branch
+                          Write-Host "git remote add fork https://github.com/azure-sdk/vcpkg -f"
+                          git remote add fork https://github.com/azure-sdk/vcpkg -f
+
+                          # If there is a remote branch by this name check it 
```

Comment:
On further review this branch preparation logic might belong in `git-branch-push.ps1` and have `git-branch-push.ps1` use an `AllowExistingPrBranchInRemote` parameter instead. 

Reason for moving this code would be that `git-branch-push.ps1` requires that this code be implicitly and if we wanted to replicate this behavior anywhere else we'd end up repeating it outside of `git-branch-push.ps1` in the other places.

Reply:
I've not looked at this in great detail but I definitely feel like you are duplicating work here that we should need to do. 

All we should need to do is clone the main repo and then make the changes and then call create-pull-request which should do everything that is needed to push to our fork and create the pull request.

Reply:
Eliminated